modules/gateways/callback/blockonomics.php

```diff
@@ -2,49 +2,46 @@
 
 // Require libraries needed for gateway module functions.
 require '../../../init.php';
-require '../../../includes/gatewayfunctions.php';
-require '../../../includes/invoicefunctions.php';
-
 require '../blockonomics/blockonomics.php';
+App::load_function('gateway');
+App::load_function('invoice');
+$gatewayParams = getGatewayVariables('blockonomics');
+if (!$gatewayParams['type']) {
+    WHMCS\Terminus::getInstance()->doDie('Module Not Activated');
+}
+
+// Check if API provides the requiered parameters
+if (
+    App::isInRequest('secret') === false ||
+    App::isInRequest('status') === false ||
+    App::isInRequest('addr') === false ||
+    App::isInRequest('value') === false
+) {
+    WHMCS\Terminus::getInstance()->doDie($_BLOCKLANG['error']['mising_params']);
+}
 
 use Blockonomics\Blockonomics;
 
 // Init Blockonomics class
 $blockonomics = new Blockonomics();
-
-$gatewayModuleName = 'blockonomics';
-
-// Fetch gateway configuration parameters.
-$gatewayParams = getGatewayVariables($gatewayModuleName);
-
-// Die if module is not active.
-if (!$gatewayParams['type']) {
-    exit('Module Not Activated');
-}
-
 require_once $blockonomics->getLangFilePath();
 
 // Retrieve data returned in payment gateway callback
-$secret = htmlspecialchars($_GET['secret']);
-$status = htmlspecialchars($_GET['status']);
-$addr = htmlspecialchars($_GET['addr']);
-$value = htmlspecialchars($_GET['value']);
-$txid = htmlspecialchars($_GET['txid']);
+$secret = App::getFromRequest('secret');
```

Comment:
This is the actual method to read queries in the URL, already sanitized and filtered by WHMCS.